### PR TITLE
[WIP] default resource limits to resource requests when limits not set

### PR DESCRIPTION
**Pull request by jsanda**
This is for #67.

### CODE REVIEW

File: pkg/apis/db/v1alpha1/cassandracluster_types_test.go [hunk 1]
```diff
@@ -457,3 +458,45 @@ func TestComputeLastAppliedConfiguration(t *testing.T) {
...
+				},
+			},
+		},
+	}
+
+	cluster.SetDefaults()
+
+	assert.Equal("orangeopensource/cassandra-image", cluster.Spec.BaseImage)
+	assert.Equal(CPUAndMem{CPU: "500m", Memory: "1Gi"}, cluster.Spec.Resources.Limits)
+}
```
Comment (cscetbon) [inline pkg/apis/db/v1alpha1/cassandracluster_types_test.go]: Let’s add tests for the other default parameters that are set, wdyt ?

Comment (jsanda) [reply pkg/apis/db/v1alpha1/cassandracluster_types_test.go]: >Let’s add tests for the other default parameters that are set, wdyt ?

👍 Absolutely!

I actually spent more time looking into validation, but will that be addressed with #34?

Comment (jsanda) [reply pkg/apis/db/v1alpha1/cassandracluster_types_test.go]: @cscetbon I update `TestSetDefaults`.  Do you want to address validation in a separate PR for #34? If so, is there anything else needed for this PR?

Comment (cscetbon) [reply pkg/apis/db/v1alpha1/cassandracluster_types_test.go]: Yes, no need to spend time here on validation. Thanks for your contribution !

**Comment by Orange-cscetbon** [comment]
@allamand I approved that PR. Please, do your magic for the e2e tests as it prevents me from merging ...

**Pull request merged**